### PR TITLE
Separate out implementaton of document-info and remove dependency of …

### PR DESCRIPTION
**Pull request by cramforce**
…3p-frame on services.

Together this improves code size for `3p-frame` based and some other extensions.

Part of #5792

### CODE REVIEW

File: src/document-info.js [hunk 1]
```diff
@@ -14,60 +14,14 @@
...
+import {getExistingServiceForDoc} from './service';
 
 
 /**
  * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
- * @return {!DocumentInfoDef} Info about the doc
+ * @return {!./service/document-info-impl.DocumentInfoDef} Info about the doc
  */
 export function documentInfoForDoc(nodeOrDoc) {
-  return /** @type {!DocumentInfoDef} */ (getServiceForDoc(nodeOrDoc,
```
Comment (jridgewell) [inline src/document-info.js]: Where's this code gone?

Comment (cramforce) [inline src/document-info.js]: I hate git.

Pushed.

File: src/service/document-info-impl.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    let canonicalUrl = rootNode && rootNode.AMP
+        && rootNode.AMP.canonicalUrl;
+    if (!canonicalUrl) {
+      const canonicalTag = rootNode.querySelector('link[rel=canonical]');
+      canonicalUrl = canonicalTag
+          ? parseUrl(canonicalTag.href).href
+          : sourceUrl;
+    }
+    const pageViewId = getPageViewId(this.ampdoc_.win);
+    return this.info_ = {url, sourceUrl, canonicalUrl, pageViewId};
```
Comment (jridgewell) [inline src/service/document-info-impl.js]: Per https://github.com/ampproject/amphtml/pull/5606#discussion_r85412412, how about not caching the `url` and `sourceUrl` attributes? This is wayy better than having a getter property.

Comment (cramforce) [inline src/service/document-info-impl.js]: Yeah, but this is a refactoring. Don't want to make any changes not related to that.

**Comment by cramforce** [comment]
Friendly ping.

**Comment by jridgewell** [comment]
@erwinmombay or me?

**Comment by erwinmombay** [comment]
💃

**Comment by cramforce** [comment]
Either or :)

**Pull request merged**